telegram/_telegramobject.py

```diff
@@ -160,37 +161,36 @@ def _get_attrs(
 
         Args:
             include_private (:obj:`bool`): Whether the result should include private variables.
-            recursive (:obj:`bool`): If :obj:`True`, will convert any TelegramObjects (if found) in
-                the attributes to a dictionary. Else, preserves it as an object itself.
+            recursive (:obj:`bool`): If :obj:`True`, will convert any ``TelegramObjects`` (if
+                found) in the attributes to a dictionary. Else, preserves it as an object itself.
             remove_bot (:obj:`bool`): Whether the bot should be included in the result.
 
         Returns:
             :obj:`dict`: A dict where the keys are attribute names and values are their values.
         """
         data = {}
 
-        if not recursive:
-            try:
-                # __dict__ has attrs from superclasses, so no need to put in the for loop below
-                data.update(self.__dict__)
-            except AttributeError:
-                pass
+        # __dict__ has attrs from superclasses, so no need to loop through them
+        if hasattr(self, "__dict__"):
+            data.update(self.__dict__)  # important when class has no __slots__.
+
```

Comment:
isn't this already covered by the below loop now? sorry for still harping on this point 😬

Reply:
haha yes. Looks like we're optimizing this even more